recipes/cfgfile/all/conanfile.py

```diff
@@ -10,82 +16,78 @@ class CfgfileConan(ConanFile):
         return {
             "Visual Studio": "15",
+            "msvc": "191",
             "gcc": "5",
             "clang": "3.5",
-            "apple-clang": "10"
+            "apple-clang": "10",
         }
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_EXAMPLES"] = False
-        self._cmake.definitions["BUILD_TESTS"] = False
-        if tools.Version(self.version) >= "0.2.9":
-            self._cmake.definitions["USE_INTERNAL_ARGS_PARSER"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+    def package_id(self):
+        del self.info.settings.compiler
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "14")
-
-        compiler = str(self.settings.compiler)
-        if compiler not in self._compilers_minimum_version:
-            self.output.warn("Unknown compiler, assuming it supports at least C++14")
-            return
-
-        version = tools.Version(self.settings.compiler.version)
-        if version < self._compilers_minimum_version[compiler]:
-            raise ConanInvalidConfiguration("cfgfile requires a compiler that supports at least C++14")
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.name} {self.version} requires C++{self._min_cppstd}, which your compiler does not support.",
+            )
 
     def requirements(self):
-        if tools.Version(self.version) >= "0.2.10":
+        if Version(self.version) >= "0.2.10":
             self.requires("args-parser/6.2.0.1")
         elif self.version == "0.2.9.1":
             self.requires("args-parser/6.2.0.1")
         elif self.version == "0.2.9.0":
             self.requires("args-parser/6.0.1.0")
 
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.0")
+
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=self.source_folder)
         cmake.build()
 
+    def layout(self):
+        cmake_layout(self)
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_EXAMPLES"] = False
+        tc.cache_variables["BUILD_TESTS"] = False
+        if Version(self.version) >= "0.2.9":
+            tc.cache_variables["USE_INTERNAL_ARGS_PARSER"] = False
```

Comment:
```suggestion
        tc.variables["BUILD_EXAMPLES"] = False
        tc.variables["BUILD_TESTS"] = False
        if Version(self.version) >= "0.2.9":
            tc.variables["USE_INTERNAL_ARGS_PARSER"] = False
```
Please use `variables` if possible.

Reply:
It's not possible in case of `cfgfile`. In my `CMakeLists.txt` options are set before `project()`, that is why I use `cache_variables`.

```cmake
cmake_minimum_required( VERSION 3.19 )

set( CFGFILE_VERSION "0.2.11" )

option( BUILD_EXAMPLES "Build examples? Default ON." ON )
option( BUILD_TESTS "Build tests? Default ON." ON )

if( NOT CMAKE_BUILD_TYPE )
	set( CMAKE_BUILD_TYPE "Release"
		CACHE STRING "Choose the type of build."
		FORCE)
endif( NOT CMAKE_BUILD_TYPE )

SET( CMAKE_CXX_STANDARD 14 )

SET( CMAKE_CXX_STANDARD_REQUIRED ON )

project( cfgfile )
```

Reply:
But with `USE_INTERNAL_ARGS_PARSER` I can use `variables`...

Reply:
@igormironchik 
Thank you for your comment!
Now I agree with you!